openwisp_users/admin.py

```diff
@@ -470,7 +470,10 @@ def save_formset(self, request, form, formset, change):
 
 
 class GroupAdmin(BaseGroupAdmin, BaseAdmin):
-    pass
+    if 'reversion' in settings.INSTALLED_APPS:
+        # Correctly register the proxy model
+        def reversion_register(self, model, **kwargs):
+            return super().reversion_register(model, for_concrete_model=False, **kwargs)
```

Comment:
@nemesisdesign I had to include this patch here since we are using a proxy model for Groups. We could instead use an Abstract `BaseGroup` class (not to be confused with `AbstractGroup`) like we are using for other auth models like User and Organization, but that'd be backward incompatible.

Also, I'm not sure how to test this in this module.

Reply:
Let's test this by adding django-reversion to requirements-test.txt and to INSTALLED_APPS in the test project, we need to make sure we won't get regressions in the future. What do you think?

Reply:
I believe we shouldn't see any regressions. From the [docs](https://django-reversion.readthedocs.io/en/stable/api.html#registration-api):
```
for_concrete_model=True
    If True proxy models will be saved under the same content type as their concrete model. If False, proxy models will be saved under their own content type, effectively giving proxy models their own distinct history.
```